### PR TITLE
Revert an erroneously removed method in the PR message builder

### PR DESCRIPTION
**Pull request by brrygrdn**
See: https://github.com/dependabot/dependabot-core/issues/2542

This method was removed in https://github.com/dependabot/dependabot-core/pull/2539 as part of a class extraction of most of the `metadata_finder(dependency)` helpers.

Unfortunately, it seems there is no test coverage that hits this code line at time of writing.

### CODE REVIEW

**Review by jurre — APPROVED** [review]
Let's improve the coverage in a follow-up but get this fix out now 👍

File: common/lib/dependabot/pull_request_creator/message_builder.rb [hunk 1]
```diff
@@ -327,6 +327,10 @@ def source_url(dependency)
         metadata_finder(dependency).source_url
       end
 
+      def homepage_url(dependency)
+        metadata_finder(dependency).homepage_url
+      end
+
```
Comment (BobbyMcWho) [inline common/lib/dependabot/pull_request_creator/message_builder.rb:333]: Does this warrant adding a test against removal of the method in the future?

**Comment by brrygrdn** [comment]
Still reading through the class, there are a few other similar `_url` helper methods that have been ✂️  - will update with those and see if I can figure out a quick test case

File: common/lib/dependabot/pull_request_creator/message_builder.rb (hunk 1 above)
Comment (brrygrdn) [reply common/lib/dependabot/pull_request_creator/message_builder.rb:337]: 👍🏻  I removed some self-stubbing on the unit-under test which was bypassing over these method calls.

File: common/spec/dependabot/pull_request_creator/message_builder_spec.rb [hunk 2]
```diff
@@ -1699,16 +1699,23 @@ def commits_details(base:, head:)
...
   end
 
-  describe "#commit_message" do
+  describe "#commit_message", :vcr do
     subject(:commit_message) { builder.commit_message }
 
-    before do
-      allow(builder).to receive(:pr_name).and_return("PR name")
-      allow(builder).to receive(:commit_message_intro).and_return("Message")
-      allow(builder).to receive(:metadata_links).and_return("\n\nLinks")
```
Comment (brrygrdn) [inline common/spec/dependabot/pull_request_creator/message_builder_spec.rb:1708]: I should have noticed these stubs meant the test wasn't exercising the real diff generation code, removing them and using VCR to capture a fixture ensures we hit the code that was broken and test for values on the `_url` helpers that were moved out of this class.

**Review by feelepxyz — APPROVED** [review]

**Pull request merged**